### ISSUE TITLE
Local Collections don't appear in Plex

### ISSUE THREAD

**Issue by hergonoway** [ZeroQI/Hama.bundle#260]
Hello, 

I'm currently struggling with Collections, so I've prepared a test folder with different kind of collections : 
- °1 :  In orange > multiple entries under an Anidb relationship (shouldn't be merge with [anidb-XXX] format)
- °2 : In Blue > Manual collection (should be merge as Collection based on parent folder name)
- °3 : In Green > Known in Scudlee files (should be merge as Collection based on scudlee files)

<img width="250" alt="image" src="https://user-images.githubusercontent.com/8405448/47956316-d79eba80-dfa2-11e8-9fbf-5923ac330c57.png">

**Result** : For now °1 and °3 behave accordingly, but I'm totally unable to generate Collection based on °2. All entries show up in Plex and all metadata is gathered.
- Unfortunatly I can't find where exactly could be logs related to Collection creation (please tell me if it exists).
- I've performed my test on a clean install of Hama bundle (with today's commit) after cleaning and restarting Plex. HTTP cache has been clean before creating the library.

Did I miss or did something wrong about Local Collection creation ? (sub-question : how can you "force" collection refresh, is it embedded in metadata refresh ?)

Here's my config file : 

```
<?xml version='1.0' encoding='utf-8'?>
<PluginPreferences>
  <rating>AniDB, TheTVDB, TheMovieDb, OMDb</rating>
  <content_rating>AniDB,TheTVDB</content_rating>
  <art>AniDB, TheTVDB, FanartTV, TheMovieDb</art>
  <EpisodeLanguagePriority>main, x-jat, en, ja</EpisodeLanguagePriority>
  <SerieLanguagePriority>main, x-jat, en, ja</SerieLanguagePriority>
  <themes>TVTunes, Plex</themes>
  <quotes>None</quotes>
  <thumbs>TheTVDB</thumbs>
  <year>AniDB</year>
  <duration>TheTVDB, AniDB, TheMovieDb, OMDb</duration>
  <rating_count>None</rating_count>
  <genres>AniDB, TheTVDB, TheMovieDb, OMDb</genres>
  <title>AniDB, TheTVDB </title>
  <MinimumWeight>3</MinimumWeight>
  <content_rating_age>None</content_rating_age>
  <writers>AniDB,TheTVDB</writers>
  <collections>Local, AnimeLists, AniDB, TheMovieDb, TheTVDB</collections>
  <absolute_index>TheTVDB</absolute_index>
  <trivia>None</trivia>
  <tags>None</tags>
  <audience_rating_image>None</audience_rating_image>
  <rating_image>None</rating_image>
  <producers>AniDB,TheTVDB</producers>
  <audience_rating>None</audience_rating>
  <studio>AnimeLists, AniDB, TheTVDB, TheMovieDb</studio>
  <adult>true</adult>
  <posters>AniDB, TheTVDB, AnimeLists, tvdb4, FanartTV, TheMovieDb</posters>
  <originally_available_at>AniDB,TheTVDB</originally_available_at>
  <roles>AniDB, TheTVDB</roles>
  <countries>TheMovieDb</countries>
  <title_sort>AniDB,TheTVDB</title_sort>
  <original_title>AniDB,TheTVDB</original_title>
  <OMDbApiKey>XXXXX</OMDbApiKey>
  <summary>AniDB, TheTVDB</summary>
  <reviews>None</reviews>
  <guest_stars>None</guest_stars>
  <directors>AniDB,TheTVDB</directors>
  <extras>None</extras>
  <tagline>TheMovieDb</tagline>
  <banners>TheTVDB</banners>
  <similar>AniDB</similar>
  <PosterLanguagePriority>ja,en,fr,ko</PosterLanguagePriority>
</PluginPreferences>
```

**Comment by ZeroQI** [comment]
Local is first in the settings field priority list.
Seem like there is a bug indeed.
I need the agent logs for that series to see Local.py output which will be in the agent data folders (see readme for location)

**Comment by EndOfLine369** [comment]
Hey @ZeroQI, looks like were generating `dict_Local` from `Local.GetMetadata` but doesn't seem like its being passed into `common.UpdateMeta`. Probably the cause.

**Comment by hergonoway** [comment]
Hi thanks for your feedback, 

I've just add a new library called "Anime - TEST8" @ 2018-11-04 11:47:36 containing a folder named "DummyFolder" with the following files : 
<img width="245" alt="image" src="https://user-images.githubusercontent.com/8405448/47963233-4888b580-e029-11e8-9d7d-294dd723ec10.png">
 

Here's the log file once done as requested : 
[com.plexapp.agents.hama.log](https://github.com/ZeroQI/Hama.bundle/files/2545748/com.plexapp.agents.hama.log)

**Comment by hergonoway** [comment]
Just updated Hama.bundle, re-run same folder, same result :(

**Comment by hergonoway** [comment]
just perform a fresh install of plex an Hama on a fresh ubuntu, set settings as mentionned above, but no collection from local folder

**Comment by EndOfLine369** [comment]
@hergonoway, still need your log files for your series not working.
[...]/Plex Media Server/Plug-in Support/Data/com.plexapp.agents.hama/DataItems/_Logs/\<library\>/\<folder\>.agent-update.log

**Comment by hergonoway** [comment]
Hello, 

PI : I've test on the last commit (710)

just create a new library, unfortunatly there no <library> named after the newly created one in the __Logs folder

i've checked at the following path : `/Plex/Library/Application Support/Plex Media Server/Plug-in Support/Data/com.plexapp.agents.hama/DataItems/_Logs/`

**Comment by EndOfLine369** [comment]
Please check your install setup. As you are missing core log files. Your os separator above indicates a unix base server your running Plex. Make sure your folder ownership for your install matches the process running your Plex process.

Your install issues aside, looked at the code and found 'collection' instead of 'collections' key string was being used. Now fixed for both AnimeLists & Local grouping folder collections.

**Comment by EndOfLine369** [comment]
@ZeroQI, looks like 'AnimeLists' is populating collection info but is not in the default pref file. Should it be added to the list at the end?
https://github.com/ZeroQI/Hama.bundle/blob/master/Contents/DefaultPrefs.json#L25
```python
FROM:
"default": "Local, AniDB, TheMovieDb, TheTVDB"
TO:
"default": "Local, AniDB, TheMovieDb, TheTVDB, AnimeLists"
```

**Comment by hergonoway** [comment]
hello retried with version from commit 711, 

all works : 
mixed Local + Animelists (as mentioned in my first post)

<img width="422" alt="image" src="https://user-images.githubusercontent.com/8405448/48316208-ce37d280-e5e0-11e8-9b77-6a3d2aae5280.png">

dumb question, should I re-add my main librairie or is there a a way to rebuild missed collection ? (metedata update, rescan, clearing stuff here and there ...) because it's gonna take weeks :D (since I'll be ban by anidb every 5 minutes for 2 days.)

**Comment by ZeroQI** [comment]
there is movie collections and series sharing the same tvdbid...
If set, and it doesn't suit the user taste, removing it will leave collection in place...
We can add it but am scared anid numbere series would be put in collection even if standalone...

**Comment by hergonoway** [comment]
@ZeroQI : sorry I'm not an English native speaker, could you rephrase or re-explain? (Sorry)

**Comment by ZeroQI** [comment]
@hergonoway can't even read it back myself... ~Not a native speaker neither...
There is a plex cache set to a day, and the agent save all succesfull xml into .\Plug-in Support\Data\com.plexapp.agents.hama\DataItems\AniDB\xml
Do you have an xml per series stored there? if not check write rights

**Comment by hergonoway** [comment]
> @hergonoway can't even read it back myself... ~Not a native speaker neither...
> There is a plex cache set to a day, and the agent save all succesfull xml into .\Plug-in Support\Data\com.plexapp.agents.hama\DataItems\AniDB\xml
> Do you have an xml per series stored there? if not check write rights

Ok so I'm not that drunk (yet) :), yes I've got an XML per show, named like that <AnidbID>.xml and all own by "plex" user with enough permissions to do whatever this user want (total control).

Since everything seems to works now regarding collections (at least for me and my options), do I have to do anything to apply this "collection update" to existing libraries ? (I mean : will existing shows ,which are currently not in any collection, be automaticaly updated and added to collections based on parameters in the agent by any automatic process and I just have to wait. Or should I'll do something manually ?)

**Comment by ZeroQI** [comment]
You can update metadata and it shouldn't query anidb again for some time if the file is there and should load it if banned even if expired. If it doesn't do that attach agent logs.

**Comment by hergonoway** [comment]
Ok so if I resume with my words : 
- Collections are updated/created with metadata update process
- Anidb cache is used (during metadata update) if the xml of the show exist in this cache (.\Plug-in Support\Data\com.plexapp.agents.hama\DataItems\AniDB\)
- If there's no corresponding file in the cache, it'll call anidb API (banned or not) to create a new cache xml file of the show
- result from anidb API calls, while banned, are not stored in this cache with an XML. (file too small is the catch)

**Comment by ZeroQI** [comment]
Exactly.
It use the file cache if recent if not it tries to download from anidb.
If not too short, it stores it.
Use the file cache even if obsolete if couldn't download...

If it doesn't do that let me know as I would need to correct code but that is how I programmed it

**Comment by hergonoway** [comment]
last dumb question, with large library (mine is around 1000 folders) could I encounter the situation were a show is never updated with anidb infos, because its at the bottom of the pile and I'll be banned half way of the daily metadata refresh ?

**Comment by ZeroQI** [comment]
The file cache allow to rotate series should be set at 2 weeks if I recall so if updating everyday it will cycle through...

**Comment by hergonoway** [comment]
OK so an Anime as a 2 week retention in anidb XML cache. Which is great. That way I can add tons of Animé even if being ban, and just wait by setting full metadata refresh to "daily" in Plex conf. 

Thanks again for your time and your answers, now I'm buffed :)

#WishListForChristmas : By the way you should create anidb-5 type which would use anidb-3or4 aggregation logic but with the "single show" provided by anidb-2 type

**Comment by ZeroQI** [comment]
Apparently, I reduced to 6 days at some point as multiple complained that the meta wasn't ready at the anime release time or shortly afterwards... launch an update every 24h and it will cycle through

I do not understand "create anidb-5 type which use anidb-3or4 aggregation logic but with the "single show" of anidb-2 type"
it creates as one tvdb entity but putting other entries as intermediate seasons...
How does it behaves for you and how do you think it should behave?

**Comment by hergonoway** [comment]
when I tested it few weeks ago it doesn't behave as it should (from what I remember) something like multiple entries in anidb4 but not in anidb2. Since then, I've update ASS which was totally out of date, and hama.bundle, repaired permissions etc ...

I'll perform a test like the one I've done in this post and create an issue if something strange comes up.

In the meantime forget what I've said :)

**Comment by hergonoway** [comment]
Well, it's seems there's a bug. I'm currently adding my whole librairy, everything seems pretty fine with series and OAV, collections building up, but something's wrong with movies.

<img width="455" alt="image" src="https://user-images.githubusercontent.com/8405448/48378428-d8d19500-e6d0-11e8-9456-0e6aba044805.png">

the 2 black cover are 2 movies, and there're added to a collection named : "Shikioriori Collection", **all other movies are also add to this collection with the same empty show** .... there's no poster, summary, almost nothing exept the Title. 
<img width="351" alt="image" src="https://user-images.githubusercontent.com/8405448/48378905-9a3cda00-e6d2-11e8-84ed-d6c3b66191f5.png">
<img width="356" alt="image" src="https://user-images.githubusercontent.com/8405448/48378921-a759c900-e6d2-11e8-9d60-5ebf1010df9a.png">


Regarding the collection it comes from a show
>> [Anidb link of Shikioriori](https://anidb.net/perl-bin/animedb.pl?show=anime&aid=13846)

here's the logs of __Logs/<MyLibrary> regarding the 2 movie in the screenshot :
[Kotonoha no Niwa [anidb-9589].agent-search.log](https://github.com/ZeroQI/Hama.bundle/files/2573865/Kotonoha.no.Niwa.anidb-9589.agent-search.log)
[Kotonoha no Niwa [anidb-9589].agent-update.log](https://github.com/ZeroQI/Hama.bundle/files/2573866/Kotonoha.no.Niwa.anidb-9589.agent-update.log)
[Sennen Joyuu [anidb-570].agent-search.log](https://github.com/ZeroQI/Hama.bundle/files/2573876/Sennen.Joyuu.anidb-570.agent-search.log)
[Sennen Joyuu [anidb-570].agent-update.log](https://github.com/ZeroQI/Hama.bundle/files/2573877/Sennen.Joyuu.anidb-570.agent-update.log)


something's wrong about the line "Inside: '['AnimeLists']'  Value: 'Shikioriori Collection'" in the logs, it's not right :)

**Comment by ZeroQI** [comment]
AniDBid: 570, url: http://api.anidb.net:9001/httpapi?request=anime&client=hama&clientver=1&protover=1&aid=570
[!] File received too small (<64 bytes), file: "<error code="500">banned</error>"

AniDBid: 9589, url: http://api.anidb.net:9001/httpapi?request=anime&client=hama&clientver=1&protover=1&aid=9589
[!] File received too small (<64 bytes), file: "<error code="500">banned</error>"

Behaving as expected considering you are temp banned on AniDB

**Comment by hergonoway** [comment]
so I'm currently unbanned, but nothing seems to change regarding Movies. I'll try to isolate Movies into a specific library and will come back with the results.

**Comment by EndOfLine369** [comment]
Fixed the collection issue of `Inside: '['AnimeLists']'  Value: 'Shikioriori Collection'`

**Comment by EndOfLine369** [comment]
Fixed the other issue seen on movie metadata not being pulled. Id numbers were not being returned for use.

**Comment by hergonoway** [comment]
Confirmed, movie metadata are gathered again, but I've still collection building issue tho.

**Comment by EndOfLine369** [comment]
FYI, also made this update to get AnimeLists collection name using lang pref. https://github.com/ZeroQI/Hama.bundle/commit/12a4d2a2d134300e17782e6bb4fa8370b038f27b

**Comment by EndOfLine369** [comment]
Also https://github.com/ZeroQI/Hama.bundle/commit/c50d828a81fe40ec626360bc49c3afbedb1e537b as TheMovieDb was removing the 'Collection' string that all the other sources were having.

**Comment by hergonoway** [comment]
well currently trying :)

for now (still upgrading metadata) it let a shit ton of empty Collection, any quick way to delete empty collection ?

**Comment by EndOfLine369** [comment]
Unfortunately no, there is no way to my knowledge to automatically delete a collection listing. That is a direct choice made by Plex to not be possible. Plex keeps all collections once created. You will have to manually delete the collections. 

Note, don't delete a collection with a series in it as that then locks the collection field (as was manually changed) and any agent meta refresh can then not update it.

**Comment by EndOfLine369** [comment]
One more fix.

- Fix  collection name split if ',' is in the name https://github.com/ZeroQI/Hama.bundle/commit/c9960898e938fed85077aaf68a155ed65955ce47
'Protect Me, Heavenly Moon Guardian! Collection' would incorrectly turn into two collections ['Protect Me', 'Heavenly Moon Guardian! Collection']

Interesting little can of worms this thread has pointed out. 😁

**Comment by EndOfLine369** [comment]
Believe everything is now fixed/enhanced as it can for collections.
All good on your end @hergonoway ?

**Comment by hergonoway** [comment]
I passed over 276 collections, nothing seems abnormal (animilist or manual collection). Only few animes (3/4) went to a collection named "Sword Gai Collection" which I don't have, but maybe I just need to wait for a clean, schedule, fresh cache, unbanned scan. I'll let few days without touching anything and go back to see that collection. I'll create a fresh issue if the problem is real. Thanks a lot for your help you can close this one !!

if I may could you answer this absolutly non related question : 
- I really can't figure what's the difference between T-EM 'title', T-EM 'title_sort' and T-EM 'original_title', could you enlight my dumb soul ?

**Comment by ZeroQI** [comment]
T-EM = Track Episode Movie
- title is what is displayed
- title_sort is what it uses for sorting, so all "The ..." are not grouped in 'T's...
- original_title is the original title like the japanese title for exemple, but Plex doesn't let agents modify it for some reason...
